src/GitHub.App/Services/ModelService.cs

```diff
@@ -366,6 +375,154 @@ IObservable<IReadOnlyList<IRemoteRepositoryModel>> GetOrganizationRepositories(s
+                            User = Create(y.Author.Login, y.Author.AvatarUrl(null))
+                        }).ToList()
+                    });
+
+                var page = await graphql.Run(query);
+                result.AddRange(page.Items);
+
+                if (page.HasNextPage)
+                    cursor = page.EndCursor;
+                else
+                    return result;
+            }
+        }
+
+        async Task<IList<IPullRequestReviewCommentModel>> GetPullRequestReviewComments(string owner, string name, int number)
+        {
+            var result = new List<IPullRequestReviewCommentModel>();
+            var query = new Query()
+                .Repository(owner, name)
+                .PullRequest(number)
+                .Reviews(first: 100, after: Var("cursor"))
+                .Select(x => new
+                {
+                    x.PageInfo.HasNextPage,
+                    x.PageInfo.EndCursor,
+                    Reviews = x.Nodes.Select(y => new
+                    {
+                        y.Id,
+                        CommentPage = y.Comments(100, null, null, null).Select(z => new
+                        {
+                            z.PageInfo.HasNextPage,
+                            z.PageInfo.EndCursor,
+                            Items = z.Nodes.Select(a => new PullRequestReviewCommentModel
+                            {
+                                Id = a.DatabaseId.Value,
+                                NodeId = a.Id,
+                                Body = a.Body,
+                                CommitId = a.Commit.Oid,
+                                CreatedAt = a.CreatedAt.Value,
+                                DiffHunk = a.DiffHunk,
+                                OriginalCommitId = a.OriginalCommit.Oid,
+                                OriginalPosition = a.OriginalPosition,
+                                Path = a.Path,
+                                Position = a.Position,
+                                PullRequestReviewId = y.DatabaseId.Value,
+                                User = Create(a.Author.Login, a.Author.AvatarUrl(null)),
+                                IsPending = y.State == Octokit.GraphQL.Model.PullRequestReviewState.Pending,
+                            }).ToList(),
+                        }).Single()
+                    }).ToList()
+                }).Compile();
+
+            var vars = new Dictionary<string, object>
+            {
+                { "cursor", null }
+            };
+
+            while (true)
+            {
+                var reviewPage = await graphql.Run(query, vars);
+
+                foreach (var review in reviewPage.Reviews)
+                {
+                    result.AddRange(review.CommentPage.Items);
+
+                    if (review.CommentPage.HasNextPage)
+                    {
+                        result.AddRange(await GetPullRequestReviewComments(review.Id, review.CommentPage.EndCursor));
```

Comment:
We should throw in some comments that explain this GraphQL a bit, because the pagination is nuanced.
Let me take a stab at it...

- The above `query` gets a single page or reviews and for each review a single page of comments.
- `GetPullRequestReviewComments` will continue to query the comments per review.
- The containing `for` loop will continue this process for each page in this page of reviews
- The containing `while` loop will continue this process to obtain all review pages

Reply:
Yep, good point - added some comments, does that make it clearer?